Framework/Geometry/test/MeshObjectTest.h

```diff
@@ -0,0 +1,1089 @@
+    triangles.push_back(2);
+    triangles.push_back(6);
+    // back face of cube - y max
+    triangles.push_back(0);
+    triangles.push_back(4);
+    triangles.push_back(1);
+    triangles.push_back(1);
+    triangles.push_back(4);
+    triangles.push_back(5);
+    // bottom face of cube - z min
+    triangles.push_back(7);
+    triangles.push_back(5);
+    triangles.push_back(6);
+    triangles.push_back(6);
+    triangles.push_back(5);
+    triangles.push_back(4);
+    // left face of cube - x min
+    triangles.push_back(7);
+    triangles.push_back(3);
+    triangles.push_back(5);
+    triangles.push_back(5);
+    triangles.push_back(3);
+    triangles.push_back(1);
+    // front fact of cube - y min
+    triangles.push_back(7);
+    triangles.push_back(6);
+    triangles.push_back(3);
+    triangles.push_back(3);
+    triangles.push_back(6);
+    triangles.push_back(2);
+
+    boost::shared_ptr<MeshObject> retVal =
+        boost::shared_ptr<MeshObject>(new MeshObject);
+    retVal->initialize(triangles, vertices);
+    return retVal;
+  }
+
+  boost::shared_ptr<MeshObject> createCube(const double size) {
+    /**
+    * Create cube of side length size with vertex at origin,
+    * parellel to axes and non-negative vertex coordinates.
+    */
+    return createCube(size, V3D(0.5 * size, 0.5 * size, 0.5 * size));
+  }
+
+  boost::shared_ptr<MeshObject> createOctahedron() {
+    /**
+    * Create octahedron with vertices on the axes at -1 & +1.
+    */
+    // The octahedron is made slightly bigger than this to
+    // ensure interior points are not rounded to be outside
+    double u = 1.0000000001;
+    std::vector<V3D> vertices;
+    vertices.push_back(V3D(u, 0, 0));
+    vertices.push_back(V3D(0, u, 0));
+    vertices.push_back(V3D(0, 0, u));
+    vertices.push_back(V3D(-u, 0, 0));
+    vertices.push_back(V3D(0, -u, 0));
+    vertices.push_back(V3D(0, 0, -u));
+
+    std::vector<int> triangles;
+    // +++ face
+    triangles.push_back(0);
+    triangles.push_back(1);
+    triangles.push_back(2);
+    //++- face
+    triangles.push_back(0);
+    triangles.push_back(5);
+    triangles.push_back(1);
+    // +-- face
+    triangles.push_back(0);
+    triangles.push_back(4);
+    triangles.push_back(5);
+    // +-+ face
+    triangles.push_back(0);
+    triangles.push_back(2);
+    triangles.push_back(4);
+    // --- face
+    triangles.push_back(3);
+    triangles.push_back(5);
+    triangles.push_back(4);
+    // --+ face
+    triangles.push_back(3);
+    triangles.push_back(4);
+    triangles.push_back(2);
+    // -++ face
+    triangles.push_back(3);
+    triangles.push_back(2);
+    triangles.push_back(1);
+    // -+- face
+    triangles.push_back(3);
+    triangles.push_back(1);
+    triangles.push_back(5);
+
+    boost::shared_ptr<MeshObject> retVal =
+        boost::shared_ptr<MeshObject>(new MeshObject);
+    retVal->initialize(triangles, vertices);
+    return retVal;
+  }
+
+  boost::shared_ptr<MeshObject> createLShape() {
```

Comment:
Similar issue as mentioned below. No need to use `shared_ptr` to wrap these objects.

Reply:
CSGObjectTest does the same thing with `createCappedCylinder()` and others.